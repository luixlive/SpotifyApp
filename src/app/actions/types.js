// Device reducer
export const DEVICE_TYPE_CHANGED = 'DEVICE_TYPE_CHANGED';

// User
export const LOAD_USER = 'LOAD_USER';
export const LOAD_USER_FAILED = 'LOAD_USER_FAILED';
export const LOAD_USER_SUCCEEDED = 'LOAD_USER_SUCCEEDED';
export const LOGOUT_USER = 'LOGOUT_USER';
export const LOGOUT_USER_FAILED = 'LOGOUT_USER_FAILED';
export const LOGOUT_USER_SUCCEEDED = 'LOGOUT_USER_SUCCEEDED';

// UserStats
export const LOAD_USER_STATS = 'LOAD_USER_STATS';
export const LOAD_USER_STATS_FAILED = 'LOAD_USER_STATS_FAILED';
export const LOAD_USER_STATS_SUCCEEDED = 'LOAD_USER_STATS_SUCCEEDED';
