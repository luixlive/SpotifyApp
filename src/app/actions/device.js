import { DEVICE_TYPE_CHANGED } from './types';

export default deviceMobile => (
  { type: DEVICE_TYPE_CHANGED, payload: { deviceMobile } }
);
