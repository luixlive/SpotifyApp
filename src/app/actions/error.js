import { CLEAR_ERROR } from './types';

export default () => (
  { type: CLEAR_ERROR, payload: {} }
);
