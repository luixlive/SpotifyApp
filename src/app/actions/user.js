import { LOAD_USER, LOGOUT_USER } from './types';

export const loadUser = () => (
  { type: LOAD_USER, payload: { } }
);

export const logoutUser = () => (
  { type: LOGOUT_USER, payload: { } }
);
