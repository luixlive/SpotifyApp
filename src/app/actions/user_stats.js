import { LOAD_USER_STATS } from './types';

export default () => (
  { type: LOAD_USER_STATS, payload: { } }
);
