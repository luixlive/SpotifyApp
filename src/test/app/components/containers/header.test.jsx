import _ from 'lodash';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

import {
  appHeaderContainer as componentProps,
} from './../../../test_utils/components_props';
import {
  ConnectedAppHeader,
  AppHeader,
} from './../../../../app/components/containers/header';
import initialState from './../../../test_utils/initial_state';
import * as types from './../../../../app/actions/types';

describe('App Components - Header', () => {
  describe('Snapshots', () => {
    let props;
    beforeEach(() => {
      props = _.cloneDeep(componentProps);
    });

    it('renders no mobile', () => {
      const rendered = renderer.create(<AppHeader {...props} />).toJSON();
      expect(rendered).toMatchSnapshot();
    });

    it('renders mobile', () => {
      props.isDeviceMobile = true;
      const rendered = renderer.create(<AppHeader {...props} />).toJSON();
      expect(rendered).toMatchSnapshot();
    });

    it('renders user authenticated', () => {
      props.isUserAuthenticated = true;
      const rendered = renderer.create(<AppHeader {...props} />).toJSON();
      expect(rendered).toMatchSnapshot();
    });
  });

  describe('Behavior', () => {
    let props;
    beforeEach(() => {
      props = _.cloneDeep(componentProps);
    });

    it('redirects to login on log in button click', () => {
      global.open = jest.fn();
      const wrapper = shallow(<AppHeader {...props} />);
      wrapper.find('Button').simulate('click');
      expect(global.open).toHaveBeenCalledTimes(1);
    });

    it('calls logoutUser on logout button click', () => {
      props.isUserAuthenticated = true;
      props.logoutUser = jest.fn();
      const wrapper = shallow(<AppHeader {...props} />);
      wrapper.find('Button').simulate('click');
      expect(props.logoutUser).toHaveBeenCalledTimes(1);
    });
  });

  describe('Provider', () => {
    const mockStore = configureStore();
    let store;
    let wrapper;
    beforeAll(() => {
      store = mockStore(initialState);
      wrapper = mount((
        <Provider store={store}>
          <ConnectedAppHeader />
        </Provider>
      ));
    });

    it('renders', () => {
      expect(wrapper.find(ConnectedAppHeader).length).toEqual(1);
    });

    it('matches initial state', () => {
      expect(Object.keys(wrapper.find(AppHeader).props()))
        .toEqual(Object.keys(componentProps));
    });

    it('dispatches logoutUser', () => {
      store = mockStore(_.merge({}, initialState, {
        user: { isUserAuthenticated: true },
      }));
      wrapper = mount((
        <Provider store={store}>
          <ConnectedAppHeader />
        </Provider>
      ));
      expect(store.getActions().length).toBe(0);
      wrapper.find('Button').simulate('click');

      const expectedAction = {
        type: types.LOGOUT_USER,
        payload: { },
      };
      expect(store.getActions()[0]).toEqual(expectedAction);
    });
  });
});
