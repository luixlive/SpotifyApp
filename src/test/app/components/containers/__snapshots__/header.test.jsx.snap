// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Components - Header Snapshots renders mobile 1`] = `
<div
  className="ui green center aligned segment"
>
  <div
    className="ui three column grid"
  >
    <div
      className="column"
    />
    <div
      className="middle aligned column"
    >
      <h3
        className="ui header"
      >
        Spotify Stats
      </h3>
    </div>
    <div
      className="column"
    >
      <button
        className="ui green inverted right floated button"
        disabled={undefined}
        onClick={[Function]}
        role="button"
        tabIndex={undefined}
      >
        Log in
      </button>
    </div>
  </div>
</div>
`;

exports[`App Components - Header Snapshots renders no mobile 1`] = `
<div
  className="ui green center aligned segment"
>
  <div
    className="ui three column grid"
  >
    <div
      className="column"
    />
    <div
      className="middle aligned column"
    >
      <h2
        className="ui header"
      >
        Spotify Stats
      </h2>
    </div>
    <div
      className="column"
    >
      <button
        className="ui green inverted right floated button"
        disabled={undefined}
        onClick={[Function]}
        role="button"
        tabIndex={undefined}
      >
        Log in with Spotify
      </button>
    </div>
  </div>
</div>
`;

exports[`App Components - Header Snapshots renders user authenticated 1`] = `
<div
  className="ui green center aligned segment"
>
  <div
    className="ui three column grid"
  >
    <div
      className="column"
    />
    <div
      className="middle aligned column"
    >
      <h2
        className="ui header"
      >
        Spotify Stats
      </h2>
    </div>
    <div
      className="column"
    >
      <button
        className="ui green inverted right floated button"
        disabled={undefined}
        onClick={[Function]}
        role="button"
        tabIndex={undefined}
      >
        Logout
      </button>
    </div>
  </div>
</div>
`;
